### ISSUE TITLE
Unclear how to subscribe to provider without a widget

### ISSUE THREAD

**Issue by filiph** [rrousselGit/riverpod#3768]
#### Describe what scenario you think is uncovered by the existing examples/articles

Say I have code such as this:

```dart
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'news.g.dart';

/// Steady and slow stream of events.
@riverpod
Stream<int> slowCounter(SlowCounterRef ref) async* {
  var i = 0;
  while (true) {
    await Future.delayed(const Duration(hours: 1));
    yield i++;
  }
}

/// Periodically fetch news from some API.
@riverpod
Future<List<Story>> fetchNews(FetchNewsRef ref) async {
  ref.watch(slowCounterProvider);
  // ... fetch stories
  return stories;
}
```

This code fetches news every hour. If I then do `ref.watch(fetchNewsProvider)` in a `HookConsumerWidget`, I can show these news in a `ListView` or whatnot.

But say that the user is _not_ looking at the news UI, but I still want to, say, show a notification about an important news story. How do I do that? Where do I listen to `fetchNews` (or `fetchNewsProvider`, rather), and how?

#### Describe why existing examples/articles do not cover this case

As far as I can tell, articles and docs only talk about listening to provider in the context of widgets, build methods, etc.

#### Additional context

Related issue from 2020: https://github.com/rrousselGit/riverpod/issues/238 "Listening to a provider without a Widget". That issue is closed with a [comment](https://github.com/rrousselGit/riverpod/issues/238#issuecomment-747922768) mentioning `ProviderListener`. But `ProviderListener` is no longer in the API (as far as I can tell).

**Comment by rrousselGit** [comment]
`ProviderListener` was replaced with `ref.listen(provider, cb)`.

The answer is more or less the same: Your app has at least one widget in it. As such, you can have a ConsumerWidget that listens to your provider.

```dart
class MyApp extends ConsumerApp {
  @override
  build(ctx, ref) {
    ref.listen(provider, (prev, next) {
      showDialog();
    });
    return ...
  }
}
```

**Comment by rrousselGit** [comment]
TL;DR, you don't "access a provider without a widget/ref". You instead find a different place where you can have a widget/ref.

**Comment by filiph** [comment]
Thanks for the quick turnaround! Understood.
